pkg/trace/metrics/metrics.go

```diff
@@ -21,6 +21,9 @@ import (
 
 // findAddr finds the correct address to connect to the Dogstatsd server.
 func findAddr(conf *config.AgentConfig) (string, error) {
+	if !conf.StatsdEnabled {
+		return "", errors.New("dogstatsd is disabled")
```

Comment:
This would make more sense at the beginning of `Configure` (line 44) instead. No need to call `findAddr` if it's disabled and we know we don't need to. Plus, I want to avoid returning an error from `Configure` if it's just disabled, since that will unnecessarily log errors and emit other startup failure events. I'm pretty sure this will actually crash the agent, too. `runAgent` in `cmd/trace-agent/subcommands/run/run.go` will return out early with an error if we return one here.

I know that you mentioned that we don't need a manual test here since it's covered by unit tests, but I don't think this case is covered. So we should either expand our unit tests and/or require manual testing.

So the code below can be something like this:

```
// Configure creates a statsd client for the given agent's configuration, using the specified global tags. If Statsd is not enabled, it will return nil and no client will be created.
func Configure(conf *config.AgentConfig, tags []string) error {
    if !conf.StatsdEnabled {
		return nil
	}
	addr, err := findAddr(conf)
	if err != nil {
		return err
	}
    ......
}
```

Reply:
The existing function already returns an error if statsd isn't otherwise configured, so I followed that pattern. Removing error returns from this function is a bit of an expansion of scope beyond the original issue.